src/Server.hs

```diff
@@ -0,0 +1,132 @@
+import           Test.QuickCheck                             (arbitrary, generate)
+
+import           ZkFold.Base.Algebra.Basic.Class             (FromConstant (..), ToConstant (..))
+import           ZkFold.Base.Algebra.Basic.Field
+import           ZkFold.Base.Algebra.EllipticCurve.BLS12_381
+import           ZkFold.Base.Algebra.EllipticCurve.Class     (EllipticCurve (ScalarField), Point)
+import           ZkFold.Base.Data.HFunctor                   (hmap)
+import           ZkFold.Base.Data.Vector
+import           ZkFold.Base.Protocol.NonInteractiveProof    (HaskellCore,
+                                                              NonInteractiveProof (Input, Proof, SetupVerify), prove,
+                                                              setupProve, setupVerify)
+import           ZkFold.Base.Protocol.Plonk                  (Plonk (Plonk))
+import           ZkFold.Base.Protocol.Plonkup.Input          (PlonkupInput (..))
+import           ZkFold.Base.Protocol.Plonkup.Proof          (PlonkupProof (..))
+import           ZkFold.Base.Protocol.Plonkup.Prover         (PlonkupProverSecret)
+import           ZkFold.Base.Protocol.Plonkup.Utils          (getParams)
+import           ZkFold.Base.Protocol.Plonkup.Witness        (PlonkupWitnessInput (PlonkupWitnessInput))
+import           ZkFold.Symbolic.Apps.KYC                    (KYCData (KYCData), kycExample)
+import           ZkFold.Symbolic.Compiler
+import           ZkFold.Symbolic.Data.Bool
+import           ZkFold.Symbolic.Data.ByteString             hiding (append, concat)
+import           ZkFold.Symbolic.Data.Combinators            (Iso (..), RegisterSize (Auto))
+import           ZkFold.Symbolic.Interpreter                 (Interpreter)
+
+type N = 18
+
+type K = 8
+
+type F = Zp BLS12_381_Scalar
+
+type Context = (Interpreter F)
+
+data InputData n k c = InputData
+  { statementId :: Int,
+    kycData     :: KYCData n k Auto c
+  }
+  deriving (Generic)
+
+instance FromJSON (InputData N K Context)
+
+instance ToJSON (InputData N K Context)
+
+data OutputData = OutputData
+  { input :: Input (PlonkKYC 29 512),
+    proof :: Proof (PlonkKYC 29 512)
+  }
+  deriving (Generic)
+
+deriving instance Generic (PlonkupInput 1 c)
+
+deriving instance Generic (PlonkupProof c)
+
+instance (ToJSON (ScalarField c), ToJSON (Point c)) => ToJSON (PlonkupProof c)
+
+instance (ToJSON (ScalarField c), ToJSON (Point c)) => ToJSON (PlonkupInput 1 c)
+
+instance (ToJSON (Point BLS12_381_G1)) => ToJSON OutputData
+
+type API = "prove" :> ReqBody '[JSON] (InputData N K Context) :> Post '[JSON] OutputData
+
+type PlonkKYC i n = Plonk U1 i n 1 BLS12_381_G1 BLS12_381_G2 BS.ByteString
+
+kycCheckVerification :: Vector 29 (ScalarField BLS12_381_G1)
+                     -> Fr
+                     -> PlonkupProverSecret BLS12_381_G1
+                     -> (SetupVerify (PlonkKYC 29 512), Input (PlonkKYC 29 512), Proof (PlonkKYC 29 512))
+kycCheckVerification witnessInputs x ps =
```

Comment:
Let's make a separate function that takes `x` and produces prover and verifier setups. I was telling @ajuggler and @zlonast before that it does not make sense that whoever calls this function has access to both `witnessInputs` and `x` since you can forge a fake proof if you just know `x`.

Reply:
I know how to fix it, here are the next steps for it:
1) save Setup constants to file instead of x, do it with `comlipeIO` (but i would prefer update it to `comlipeIOWith`)
2) read Setup from json, and after that construct the Proof
¯\\\_(ツ)\_/¯